{
    "body": "Previously, if an error occurred during untarring, Spack would crash due to a missing `spack-build.out` file. Now, it exits a little more gracefully, revealing the real error message.\r\n\r\n### Before\r\n\r\n```console\r\n$ spack install gmp\r\n==> Installing gmp\r\n==> Using cached archive: /home/hpc/spack/var/spack/cache/gmp/gmp-6.1.2.tar.bz2\r\n==> Staging archive: /home/hpc/spack/var/spack/stage/gmp-6.1.2-6bsovvkqwx6zscwtbvjj6egrgizbyycm/gmp-6.1.2.tar.bz2\r\ntar (child): lbzip2: Cannot exec: No such file or directory\r\ntar (child): Error is not recoverable: exiting now\r\n/usr/bin/tar: Child returned status 2\r\n/usr/bin/tar: Error is not recoverable: exiting now\r\n==> Error: ProcessError: Command exited with status 2:\r\n   '/usr/bin/tar' '-xf' '/home/hpc/spack/var/spack/stage/gmp-6.1.2-6bsovvkqwx6zscwtbvjj6egrgizbyycm/gmp-6.1.2.tar.bz2'\r\n==> Error: [Errno 2] No such file or directory: '/home/hpc/spack/var/spack/stage/gmp-6.1.2-6bsovvkqwx6zscwtbvjj6egrgizbyycm/spack-expanded-archive/spack-build.out'\r\n```\r\n```console\r\n$ spack --debug install gmp\r\n==> Installing gmp\r\n==> Deprecation warning: md5 checksums will not be supported in future Spack releases.\r\n==> Using cached archive: /home/hpc/spack/var/spack/cache/gmp/gmp-6.1.2.tar.bz2\r\n==> Deprecation warning: md5 checksums will not be supported in future Spack releases.\r\n==> Staging archive: /home/hpc/spack/var/spack/stage/gmp-6.1.2-6bsovvkqwx6zscwtbvjj6egrgizbyycm/gmp-6.1.2.tar.bz2\r\n==> '/usr/bin/tar' '-xf' '/home/hpc/spack/var/spack/stage/gmp-6.1.2-6bsovvkqwx6zscwtbvjj6egrgizbyycm/gmp-6.1.2.tar.bz2'\r\ntar (child): lbzip2: Cannot exec: No such file or directory\r\ntar (child): Error is not recoverable: exiting now\r\n/usr/bin/tar: Child returned status 2\r\n/usr/bin/tar: Error is not recoverable: exiting now\r\n==> Error: ProcessError: Command exited with status 2:\r\n   '/usr/bin/tar' '-xf' '/home/hpc/spack/var/spack/stage/gmp-6.1.2-6bsovvkqwx6zscwtbvjj6egrgizbyycm/gmp-6.1.2.tar.bz2'\r\nTraceback (most recent call last):\r\n File \"/home/hpc/spack/bin/spack\", line 61, in <module>\r\n   sys.exit(spack.main.main())\r\n File \"/home/hpc/spack/lib/spack/spack/main.py\", line 653, in main\r\n   return _invoke_command(command, parser, args, unknown)\r\n File \"/home/hpc/spack/lib/spack/spack/main.py\", line 432, in _invoke_command\r\n   return_val = command(parser, args)\r\n File \"/home/hpc/spack/lib/spack/spack/cmd/install.py\", line 273, in install\r\n   install_spec(args, kwargs, spec)\r\n File \"/home/hpc/spack/lib/spack/spack/cmd/install.py\", line 154, in install_spec\r\n   spec.package.do_install(**kwargs)\r\n File \"/home/hpc/spack/lib/spack/spack/package.py\", line 1495, in do_install\r\n   self, build_process, dirty=dirty, fake=fake)\r\n File \"/home/hpc/spack/lib/spack/spack/build_environment.py\", line 776, in fork\r\n   child_result.print_context()\r\n File \"/home/hpc/spack/lib/spack/spack/error.py\", line 75, in print_context\r\n   if self.long_message:\r\n File \"/home/hpc/spack/lib/spack/spack/build_environment.py\", line 912, in long_message\r\n   errors, warnings = parse_log_events(self.build_log)\r\n File \"/home/hpc/spack/lib/spack/spack/util/log_parse.py\", line 58, in parse_log_events\r\n   result = parse_log_events.ctest_parser.parse(stream, context, jobs)\r\n File \"/home/hpc/spack/lib/spack/external/ctest_log_parser.py\", line 426, in parse\r\n   with open(stream) as f:\r\nIOError: [Errno 2] No such file or directory: '/home/hpc/spack/var/spack/stage/gmp-6.1.2-6bsovvkqwx6zscwtbvjj6egrgizbyycm/spack-expanded-archive/spack-build.out\r\n```\r\n### After\r\n```console\r\n$ spack install gmp\r\n==> Installing gmp\r\n==> Using cached archive: /home/hpc/spack/var/spack/cache/gmp/gmp-6.1.2.tar.bz2\r\n==> Staging archive: /home/hpc/spack/var/spack/stage/gmp-6.1.2-6bsovvkqwx6zscwtbvjj6egrgizbyycm/gmp-6.1.2.tar.bz2\r\ntar (child): lbzip2: Cannot exec: No such file or directory\r\ntar (child): Error is not recoverable: exiting now\r\n/usr/bin/tar: Child returned status 2\r\n/usr/bin/tar: Error is not recoverable: exiting now\r\n==> Error: ProcessError: Command exited with status 2:\r\n   '/usr/bin/tar' '-xf' '/home/hpc/spack/var/spack/stage/gmp-6.1.2-6bsovvkqwx6zscwtbvjj6egrgizbyycm/gmp-6.1.2.tar.bz2'\r\n```\r\nI'm not sure if there are other places where we try to use `self.build_log` without first checking if it exists.\r\n\r\nReported on Slack by Strahinja Trecakov.",
    "user": "adamjstewart",
    "url": "https://api.github.com/repos/spack/spack/issues/8981",
    "updated_at": "2018-08-21 22:13:03",
    "created_at": "2018-08-14 21:13:57",
    "closed_at": "2018-08-21 21:16:52",
    "state": "closed",
    "title": "Fix error caused by missing spack-build.out build log",
    "number": 8981,
    "milestone": null,
    "labels": [
        "error-messages"
    ],
    "id": 350600216,
    "html_url": "https://github.com/spack/spack/pull/8981",
    "assignees": [],
    "comments": 0
}